mmdet3d/core/bbox/coders/km3d_decoder.py

```diff
@@ -0,0 +1,383 @@
+        .float()  # boxes 2 3
+    kps = kps.view(batch, boxes, -1, 2).permute(0, 1, 3, 2)  # batch boxes 2 9
+    hom = kps.new_ones(batch, boxes, 1, 9)  # batch boxes 1 9
+    kps = torch.cat((kps, hom), dim=2).view(-1, 3, 9)  # boxes 3 9
+
+    # batch boxes 2 9     [[xxxxxxx...],[yyyyyy...]]
+    kps = torch.bmm(out_inv, kps).view(batch, boxes, 2, 9)
+    # batch boxes 18 [xyxyxyxyx....]
+    kps = kps.permute(0, 1, 3, 2).view(batch, boxes, -1)
+    # calib2 batch boxes 1
+    calib2 = torch.zeros_like(kps[:, :, 0:1]) + calib[:, 0:1, 0:1]
+    alpha_idx = rot[:, :, 1] > rot[:, :, 5]  # batch boxes
+    alpha_idx = alpha_idx.float()  # batch boxes
+    # alpha1 Shape: [batch boxes]
+    alpha1 = torch.atan(rot[:, :, 2] / rot[:, :, 3]) + (-0.5 * np.pi)
+    alpha2 = torch.atan(rot[:, :, 6] / rot[:, :, 7]) + (0.5 * np.pi)
+    alpna_pre = alpha1 * alpha_idx + alpha2 * (1 - alpha_idx)  # batch boxes
+    alpna_pre = alpna_pre.unsqueeze(2)  # batch boxes
+
+    # batch boxes 1
+    rot_y = alpna_pre + torch.atan2(kps[:, :, 16:17] - calib[:, 0:1, 2:3],
+                                    calib2)
+    rot_y[rot_y > np.pi] = rot_y[rot_y > np.pi] - 2 * np.pi  # batch boxes 1
+    rot_y[rot_y < -np.pi] = rot_y[rot_y < -np.pi] + 2 * np.pi
+
+    calib = calib.unsqueeze(1)  # 1 1 3 4
+    calib = calib.expand(batch, boxes, -1, -1).contiguous()  # batch boxes 3 4
+    kpoint = kps[:, :, :16]  # batch boxes 16
+    f = calib[:, :, 0, 0].unsqueeze(2)  # batch boxes 1
+    f = f.expand_as(kpoint)  # batch boxes 16
+    # batch boxes 1 , batch boxes 1
+    cx, cy = calib[:, :, 0, 2].unsqueeze(2), calib[:, :, 1, 2].unsqueeze(2)
+    cxy = torch.cat((cx, cy), dim=2)  # 1 boxes 2
+    cxy = cxy.repeat(1, 1, 8)  # 1 boxes 16
+    kp_norm = (kpoint - cxy) / f  # 1 boxes 16
+
+    height = dim[:, :, 0:1]  # batch boxes 1
+    width = dim[:, :, 1:2]  # batch boxes 1
+    length = dim[:, :, 2:3]  # batch boxes 1
+    cos_rot_y = torch.cos(rot_y)  # batch boxes 1
+    sin_rot_y = torch.sin(rot_y)  # batch boxes 1
+
+    matrix_B = torch.zeros_like(kpoint)  # batch boxes 16
+    matrix_C = torch.zeros_like(kpoint)  # batch boxes 16
+
+    kp = kp_norm.unsqueeze(3)  # batch,boxes,16,1
+    const = const.expand(batch, boxes, -1, -1)  # batch boxes 16 2
+    matrix_A = torch.cat([const, kp], dim=3)  # batch boxes 16 3
+
+    data_B = torch.FloatTensor([
+        length * 0.5 * cos_rot_y + width * 0.5 * sin_rot_y, height * 0.5,
+        length * 0.5 * cos_rot_y - width * 0.5 * sin_rot_y, height * 0.5,
+        -length * 0.5 * cos_rot_y - width * 0.5 * sin_rot_y, height * 0.5,
+        -length * 0.5 * cos_rot_y + width * 0.5 * sin_rot_y, height * 0.5,
+        length * 0.5 * cos_rot_y + width * 0.5 * sin_rot_y, -height * 0.5,
+        length * 0.5 * cos_rot_y - width * 0.5 * sin_rot_y, -height * 0.5,
+        -length * 0.5 * cos_rot_y - width * 0.5 * sin_rot_y, -height * 0.5,
+        -length * 0.5 * cos_rot_y + width * 0.5 * sin_rot_y, -height * 0.5
+    ])
```

Comment:
Refer to the implementation of this [function](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/core/bbox/structures/lidar_box3d.py#L49)

Also, the data_B, matrix_B has no actual meanings. Please rename these variables.